api/v1/logstorage_types.go

```diff
@@ -42,6 +42,11 @@ type LogStorageSpec struct {
 	// +optional
 	StorageClassName string `json:"storageClassName,omitempty"`
 
+	// DataAffinity allows configuration of the affinity characteristics for Elasticsearch to provide more control over
+	// where Elasticsearch will run. The contents of DataAffinity will be added to the PodSpec of the Elasticsearch nodes.
+	// +optional
+	DataAffinity *corev1.Affinity `json:"dataAffinity,omitempty"`
```

Comment:
I think you should look at the NodeSets definition. We already have affinity for elasticsearch it's just done a little differently. It also ensures that index replicas are spread across the nodes.

Reply:
I see. So we should remove this since affinity for Elasticsearch is handled already.